.github/workflows/main.yml

```diff
@@ -55,7 +55,7 @@ jobs:
         # Not utilizing makefile target here to make use of pipefail bash feature.
         run: |
           set -euo pipefail
-          go test -p 1 -json ./... -covermode=count -coverprofile=coverage.out 2>&1 | tee test_results.json
+          GODEBUG=netdns=cgo go test -p 1 -json ./... -covermode=count -coverprofile=coverage.out 2>&1 | tee test_results.json
```

Comment:
This didn't seem to have any effect on the new test failures:

```
=== RUN   TestPeerMultiAddrFromServiceURL_Success
=== RUN   TestPeerMultiAddrFromServiceURL_Success/IPv6
    url_conversion_test.go:38: 
        	Error Trace:	/home/runner/work/pocket/pocket/libp2p/network/url_conversion_test.go:38
        	Error:      	Received unexpected error:
        	            	resolving peer IP for hostname: 2a00:1450:4005:802::2004: address 2a00:1450:4005:802::2004: no suitable address found
        	Test:       	TestPeerMultiAddrFromServiceURL_Success/IPv6
--- FAIL: TestPeerMultiAddrFromServiceURL_Success (0.00s)
    --- FAIL: TestPeerMultiAddrFromServiceURL_Success/IPv6 (0.00s)
=== RUN   TestPeerMultiAddrFromServiceURL_Error
=== RUN   TestPeerMultiAddrFromServiceURL_Error/fully_qualified_domain_name_with_scheme/ip6
    url_conversion_test.go:100: 
        	Error Trace:	/home/runner/work/pocket/pocket/libp2p/network/url_conversion_test.go:100
        	Error:      	Error "resolving peer IP for hostname: tcp: lookup tcp: Temporary failure in name resolution" does not contain "no such host"
        	Test:       	TestPeerMultiAddrFromServiceURL_Error/fully_qualified_domain_name_with_scheme/ip6
        	Messages:   	actualMultiaddr: %!s(<nil>)
=== RUN   TestPeerMultiAddrFromServiceURL_Error/fully_qualified_domain_name_with_scheme/fqdn
    url_conversion_test.go:100: 
        	Error Trace:	/home/runner/work/pocket/pocket/libp2p/network/url_conversion_test.go:100
        	Error:      	Error "resolving peer IP for hostname: tcp: lookup tcp: Temporary failure in name resolution" does not contain "no such host"
        	Test:       	TestPeerMultiAddrFromServiceURL_Error/fully_qualified_domain_name_with_scheme/fqdn
        	Messages:   	actualMultiaddr: %!s(<nil>)
=== RUN   TestPeerMultiAddrFromServiceURL_Error/fully_qualified_domain_name_with_scheme/ip4
    url_conversion_test.go:100: 
        	Error Trace:	/home/runner/work/pocket/pocket/libp2p/network/url_conversion_test.go:100
        	Error:      	Error "resolving peer IP for hostname: tcp: lookup tcp: Temporary failure in name resolution" does not contain "no such host"
        	Test:       	TestPeerMultiAddrFromServiceURL_Error/fully_qualified_domain_name_with_scheme/ip4
        	Messages:   	actualMultiaddr: %!s(<nil>)
=== RUN   TestPeerMultiAddrFromServiceURL_Error/missing_port_number/ip6
    url_conversion_test.go:100: 
        	Error Trace:	/home/runner/work/pocket/pocket/libp2p/network/url_conversion_test.go:100
        	Error:      	Error "resolving peer IP for hostname: 2a00:1450:4005:802::2004: address 2a00:1450:4005:802::2004: no suitable address found" does not contain "unexpected end of multiaddr"
        	Test:       	TestPeerMultiAddrFromServiceURL_Error/missing_port_number/ip6
        	Messages:   	actualMultiaddr: %!s(<nil>)
--- FAIL: TestPeerMultiAddrFromServiceURL_Error (0.02s)
    --- FAIL: TestPeerMultiAddrFromServiceURL_Error/fully_qualified_domain_name_with_scheme/ip6 (0.01s)
    --- FAIL: TestPeerMultiAddrFromServiceURL_Error/fully_qualified_domain_name_with_scheme/fqdn (0.01s)
    --- FAIL: TestPeerMultiAddrFromServiceURL_Error/fully_qualified_domain_name_with_scheme/ip4 (0.01s)
    --- FAIL: TestPeerMultiAddrFromServiceURL_Error/missing_port_number/ip6 (0.00s)
```

Reply:
I think there are a couple of things happening. First one is ipv6 being a special case, and a second one might be related to different errors on different platforms. This is just a hypothesis at this point and I need to look deeper. Do you run linux on your dev machine?

I seem to be only able to replicate ipv6 error locally and I am on mac, but I should mention my did not set up ipv6 at home.
```
❯ go test -count=1 ./libp2p/...
--- FAIL: TestPeerMultiAddrFromServiceURL_Success (0.00s)
    --- FAIL: TestPeerMultiAddrFromServiceURL_Success/IPv6 (0.00s)
        url_conversion_test.go:38:
            	Error Trace:	/Users/dk/pocket/pocket/libp2p/network/url_conversion_test.go:38
            	Error:      	Received unexpected error:
            	            	resolving peer IP for hostname: 2a00:1450:4005:802::2004: address 2a00:1450:4005:802::2004: no suitable address found
            	Test:       	TestPeerMultiAddrFromServiceURL_Success/IPv6
--- FAIL: TestPeerMultiAddrFromServiceURL_Error (15.02s)
    --- FAIL: TestPeerMultiAddrFromServiceURL_Error/missing_port_number/ip6 (0.00s)
        url_conversion_test.go:100:
            	Error Trace:	/Users/dk/pocket/pocket/libp2p/network/url_conversion_test.go:100
            	Error:      	Error "resolving peer IP for hostname: 2a00:1450:4005:802::2004: address 2a00:1450:4005:802::2004: no suitable address found" does not contain "unexpected end of multiaddr"
            	Test:       	TestPeerMultiAddrFromServiceURL_Error/missing_port_number/ip6
            	Messages:   	actualMultiaddr: %!s(<nil>)
FAIL
FAIL	github.com/pokt-network/pocket/libp2p/network	15.207s
?   	github.com/pokt-network/pocket/libp2p/transport	[no test files]
?   	github.com/pokt-network/pocket/libp2p/types/mocks	[no test files]
FAIL
```

Now, the other errors - if you read that - "resolving peer IP for hostname: tcp: lookup tcp: Temporary failure in name resolution" you can see it tries to resolve hostname `tcp` but Golang/stdlib returns `Temporary failure in name resolution` instead of `no such host`.. I suspect this happens due to inability to lookup a domain `tcp` (special case?) or the error is just different on Linux platform. I'm happy to validate/test this on fresh Linux box on Monday.

Would it be a good idea to detect ipv6/ipv6 addresses and skip DNS resolve for them? And only perform DNS request on hostnames.

Reply:
I can confirm that I'm seeing the same thing as @okdas when running this locally.

<img width="1494" alt="Screenshot 2023-02-24 at 4 07 14 PM" src="https://user-images.githubusercontent.com/1892194/221323621-2cff55d7-8c51-4aab-aa5e-a5f5fee92b49.png">

---

@bryanchriswhite does this work on your local machine?

Reply:
Yes, those tests have been consistently passing locally on my Linux workstation.

Reply:
EDIT: there were two tests that I apparently broke at some point and didn't notice.